mountpoint-s3-client/src/s3_crt_client/put_object.rs

```diff
@@ -206,7 +194,66 @@ impl PutObjectRequest for S3PutObjectRequest {
         let elapsed = self.start_time.elapsed();
         emit_throughput_metric(self.total_bytes, elapsed, "put_object");
 
-        check_response_headers(self.response_headers, &self.server_side_encryption, &self.ssekms_key_id);
+        let locked_value = self
```

Comment:
tiny nit: rename or inline in next statement

Reply:
inlined the whole thing to the function call, a single temporary variable does not work:
```
        let response_headers = self
            .response_headers
            .lock()
            .expect("must be able to acquire headers lock") // <--- temporary value dropped while borrowed
            .as_ref()
            .expect("PUT response headers must be available at this point");
        check_response_headers(
            response_headers,
            self.server_side_encryption.as_deref(),
            self.ssekms_key_id.as_deref(),
        );
```

Reply:
That's okay. FYI: in this case you are consuming `self` (and already assume you can unwrap), so you could also have extracted the value from the Arc<Mutex<..>>, rather than just referencing it.